Data Prep Code.Rmd

```diff
@@ -0,0 +1,317 @@
+
+# Store the p-value for use in the plot annotation
+p_value <- round(kruskal_test$p.value, 4)
+
+# Create an annotation text with the Kruskal-Wallis result
+annotation_text <- paste("Kruskal-Wallis, p =", p_value, ". Note. Confidence intervals on means are at 95%.")
+print(annotation_text)
+
+# Plot the bar graph with sorted bars
+ggplot(bully_dictator_game_data, aes(x = treatment_group, y = Mean_Got, fill = treatment_group)) +
+  geom_bar(stat = "identity", width = 0.5, alpha = 0.5) +
+  geom_errorbar(aes(ymin = Mean_Got - SE, ymax = Mean_Got + SE), width = 0.2, color = "black") +
+  scale_fill_manual(values = c("C" = "red", "TD" = "green", "MD" = "blue")) +
+  labs(title = "Figure 3: Bully Dictator Game", y = "Mean of Got", x = NULL) +
+  theme_minimal() +
+  theme(plot.title = element_text(hjust = 0.5), legend.position = "none")
+  
+```
+
+
+```{r Figure3 MW KW, echo=FALSE}
+# Perform Kruskal–Wallis test for DG
+# Testing for differences in the 'kept' variable distributions across different levels of 'moc' within the Standard Dictator Game
+kruskal_DG <- kruskal.test(kept ~ moc, data = DG)
+print("Kruskal-Wallis test for Standard Dictator Game:")
+print(kruskal_DG)
+
+# Perform Kruskal–Wallis test for BDG
+# Testing for differences in the 'kept' variable distributions across different levels of 'moc' within the Bully Dictator Game
+kruskal_BDG <- kruskal.test(kept ~ moc, data = BDG)
+print("Kruskal-Wallis test for Bully Dictator Game:")
+print(kruskal_BDG)
+
+# Perform Mann-Whitney U test for BDG choice
+group_C <- BDG[BDG$treatment_group == "C", "kept"]$kept
+group_TD <- BDG[BDG$treatment_group == "TD", "kept"]$kept
+group_MD <- BDG[BDG$treatment_group == "MD", "kept"]$kept
+
+# Perform Mann-Whitney U test between TD and MD for BDG
+# This test compares the distribution of 'kept' between TD and MD within the Bully Dictator Game
+result_TD_MD <- wilcox.test(group_TD, group_MD)
+z_value_TD_MD <- qnorm(1 - result_TD_MD$p.value / 2)  # Convert p-value to z-score
+p_value_TD_MD <- result_TD_MD$p.value
+cat(sprintf("Mann-Whitney U test for BDG between TD and MD, z = %.3f, p = %.4f.\n", z_value_TD_MD, p_value_TD_MD))
+
+# Perform Mann-Whitney U test between C and MD for BDG
+# This test compares the distribution of 'kept' between C and MD within the Bully Dictator Game
+result_C_MD <- wilcox.test(group_C, group_MD)
+z_value_C_MD <- qnorm(1 - result_C_MD$p.value / 2)
+p_value_C_MD <- result_C_MD$p.value
+cat(sprintf("Mann-Whitney U test for BDG between C and MD, z = %.3f, p = %.4f.\n", z_value_C_MD, p_value_C_MD))
+
+# Perform Mann-Whitney U test between C and TD for BDG
+# This test compares the distribution of 'kept' between C and TD within the Bully Dictator Game
+result_C_TD <- wilcox.test(group_TD, group_C)
+z_value_C_TD <- qnorm(1 - result_C_TD$p.value / 2)
+p_value_C_TD <- result_C_TD$p.value
+cat(sprintf("Mann-Whitney U test for BDG between C and TD, z = %.3f, p = %.4f.\n", z_value_C_TD, p_value_C_TD))

```

Comment:
Please use separate chunks for printing text/data/plots and write a comments as a plain text between those chunks.

Reply:
done